src/helpers/createPdfAndSave.js

```diff
@@ -0,0 +1,154 @@
+import { PDFDocument } from 'pdf-lib'
+
+import { models } from 'cozy-client'
+
+import { CONTACTS_DOCTYPE, FILES_DOCTYPE } from 'src/doctypes'
+import { addFileToPdf } from 'src/utils/addFileToPdf'
+import { buildFilename } from 'src/helpers/buildFilename'
+
+const {
+  contact: { getFullname },
+  file: { uploadFileWithConflictStrategy }
+} = models
+
+/**
+ * @typedef {object} SaveFileParam
+ * @property {CozyClient} client
+ * @property {DocumentCollection} fileCollection
+ * @property {ArrayBuffer} pdfBytes
+ * @property {string} contactId
+ * @property {{ name: string, metadata: object, dirId: string }} opts
+ */
+
+/**
+ * @typedef {object} createAndSavePdfParam
+ * @property {{ data: object[], metadata: object }} formData
+ * @property {Qualification} qualification
+ * @property {Paper} currentDefinition
+ * @property {string} appFolderID
+ * @property {CozyClient} client
+ * @property {{t: Function, f: Function, scannerT: Function}} i18n
+ */
+
+/**
+ * @param {{multipage: boolean, page: string}} fileMetadata
+ * @param {boolean} isMultipage
+ * @returns {object}
+ */
+const sanitizeFileMetadata = (fileMetadata, isMultipage) => {
+  if (isMultipage || !fileMetadata.page) return {}
+
+  // eslint-disable-next-line no-unused-vars
+  const { multipage, ...newFileMetadata } = fileMetadata
+
+  return newFileMetadata
+}
+
+/**
+ * @param {SaveFileParam} param
+ * @returns {Promise<string>} Filename
+ */
+const saveFile = async ({
+  client,
+  fileCollection,
+  pdfBytes,
+  contacts,
+  opts
+}) => {
+  const { name, metadata, dirId } = opts
+  const { data: fileCreated } = await uploadFileWithConflictStrategy(
+    client,
+    pdfBytes,
+    {
+      name,
+      contentType: 'application/pdf',
+      metadata,
+      dirId,
+      conflictStrategy: 'rename'
+    }
+  )
+
+  // The user is referenced as a contact on the new Paper
+  const references = contacts.map(contact => ({
+    _id: contact._id,
+    _type: CONTACTS_DOCTYPE
+  }))
+  await fileCollection.addReferencedBy(fileCreated, references)
```

Comment:
no need to pass it as an argument. We already have the client you can build it from there